src/Halogen/Driver.purs

```diff
@@ -131,7 +131,7 @@ runUI c s element = _.driver <$> do
         when (isJust rp) $ render ref
         putVar rpRef Nothing
         q
-      HaltHF -> empty
+      HaltHF msg -> throwError $ error msg
```

Comment:
Thinking about it some more, I think the only reason exceptions during rendering nuked the driver was because it results in an `AVar` not being put back, creating a dead lock.